src/mbed_tools/build/_internal/config/assemble_build_config.py

```diff
@@ -47,6 +60,20 @@ def _assemble_config_from_sources_and_lib_files(
     return Config.from_sources(all_sources)
 
 
+def _apply_requires_filter(requires_filtered_files: set, files: Iterable[Path], requires: list) -> None:
+    """Remove mbed_lib.json files not required by application.
+
+    The 'requires' config option in mbed_app.json can specify list of mbed
+    libraries (mbed_lib.json) that application requires. Apply 'requires'
+    filter to remove mbed_lib.json files not required by application.
+    """
+    for required_mbed_lib in requires:
+        for mbed_lib_path in files:
+            if required_mbed_lib == _get_mbed_lib_name(mbed_lib_path):
+                requires_filtered_files.add(mbed_lib_path)
+                _apply_requires_filter(requires_filtered_files, files, _get_mbed_lib_requires(mbed_lib_path))
```

Comment:
Is it possible to just extend `_filter_files` with another filter type, instead of introducing this extra filtering mechanism?

Reply:
Updated `_filter_files` with requires filter type and moved `_apply_requires_filter` requires filter.